Document getting started as a developer (#689)
steveicarus/iverilog: Documentation/developer/getting_started.rst

Getting Started as a Contributer
================================

Icarus Verilog development is centered around the github repository at
`github.com/steveicarus/iverilog <http://github.com/steveicarus/iverilog>`_.
Contributing to Icarus Verilog requires a basic knowledge of git and github,
so see the github documentation for more information. The sections below will
step you through the basics of getting the source code from github, making a
branch, and submitting a pull request for review.

Getting Icarus Verilog
----------------------

To start, you will need to clone the code. It is preferred that you use the
"ssh" method, and the ssh based clone with the command:

> .. code-block:: none

Reviewer: ```suggestion
.. code-block:: console
```

To indicate that this is an interactive shell session.

Author: Didn't know that was an option. Done.

most people will want to do.) You should rerun this test before submitting
patches to the developers. Also, if you are adding a new feature, you should
add test programs to the regression test suite to validate your new feature
(or bug fix.)

Note that pull requests will be required to pass these regression tests before
being merged.

Forks, Branches and Pull Requests
---------------------------------

Currently, the preferred way to submit patches to Icarus Verilog is via pull
requests.
`Pull requests <https://docs.github.com/en/github-ae@latest/pull-requests>`_
can be created from the main repository if you have write access (very few
people have write access) or more commonly from a fork, so the first step is
to create a fork that you can work with. It is easy enough to create a fork,
just go to the
`github.com/steveicarus/iverilog <http://github.com/steveicarus/iverilog>`_
page and use the "fork" button in the upper right corner. This will create
a new repository that you can clone instead of the steveicarus/iverilog
repository. You then use your local repository to create feature branches,
then submit them for inclusion in the main repository as pull

> requests. Remember to use the "Fetch upstream" feature periodically to keep
> your fork synchronized with the main repository.

Reviewer: ```suggestion
requests. Remember to `synchronize your fork
<https://docs.github.com/en/github-ae@latest/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork>`_
periodically with the main repository. This will make sure your work is based on
the latest upstream and avoid merge conflicts.
```

There are a few different methods available for syncing, not just the "Fetch upstream" button.

Author: Got it.

Create your patch by first creating a branch that contains your commits:

.. code-block:: none

>   git checkout -b my-github-id/branch-name

Reviewer: All other command lines have a `%`.
```suggestion
  % git checkout -b my-github-id/branch-name
```

Although I'm not sure if it should be there in the first place since it makes it more difficult to copy.

Author: Oops., Fixed.
